recipe/build-pkg.sh

```diff
@@ -1,15 +1,31 @@
-CUDA_CONFIG_ARG=""
+set -e
+
+FAISS_ENABLE_GPU=""
 if [ ${cuda_compiler_version} != "None" ]; then
-    CUDA_CONFIG_ARG="--with-cuda=${CUDA_HOME}"
+    FAISS_ENABLE_GPU="ON"
 else
-    CUDA_CONFIG_ARG="--without-cuda"
+    FAISS_ENABLE_GPU="OFF"
 fi
 
-# Build vanilla version (no avx)
-./configure --with-blas=-lblas --with-lapack=-llapack ${CUDA_CONFIG_ARG}
+if [ ${osx} ]; then
+    # cmake detection on OSX doesn't work out of the box
+    OSX_ARGS = "-DPython_EXECUTABLE=$PYTHON -DPython_LIBRARIES=$LIBRARY_LIB -DPython_INCLUDE_DIR=$LIBRARY_INC"
```

Comment:
- I think `LIBRARY_LIB`/`LIBRARY_INC` are only set on Windows.
- Acc. to https://cmake.org/cmake/help/latest/module/FindPythonLibs.html you'd want `Python_LIBRARY` instead of `Python_LIBRARIES`.
- I'm not sure, but I'd think `Python_LIBRARY` has to point to the shared library file itself and `Python_INCLUDE_DIR` has to point to `include/python*` instead of just `include`.

IDK, but maybe setting `Python_EXECUTABLE` suffices; that would solve/circumvent setting the `LIBRARY`/`INCLUDE_DIR` values too.

Reply:
I've tried that without success...

Reply:
I've added a commit to try if setting only `Python_EXECUTABLE` works.

Reply:
> I've tried that without success...

Ah, okay. Let me see what the CI spills out with the current commit to see how it behaves.

Reply:
It spat out precisely the lack of lib & include, which is why I added in the first place (btw, I really hate that azure deletes logs for runs where there are jobs that didn't fully complete, otherwise I could already directly refer you to the CI of a commit above: https://github.com/conda-forge/faiss-split-feedstock/runs/1506991663)

Reply:
Might be some weird interaction with the cmake-cache - now it works. On second thought, I think things generally failed on whatever python build was second...

Reply:
> btw, I really hate that azure deletes logs for runs where there are jobs that didn't fully complete

Yup, same :(. Also, the short retention time (although we already upped them from their even shorter default) can be annoying when revisiting PRs that have been open for a while :/...


----
> now it works.

Let's just hope we don't hit some weird indeterminism and the next build doesn't fail again XD.
So, if you've adventurous (or rather not already totally fed up by this PR ;P) you can try again without the `Python_EXECUTABLE` to see if CMake's default also work when it has a clean cache. (But it's really not important -- enjoying a Sunday evening is probably a better investment :).)

Reply:
As soon as windows is done failing (so the logs persists), I'll push some more work.